ctl/src/system_info.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::{anyhow, Context, Result};
+use serde_derive::Deserialize;
+use std::env;
+
+#[derive(Deserialize)]
+struct SystemInfo {
+    git_sha: String,
+    git_commit_date: String,
+}
+
+fn read_system_info() -> Result<SystemInfo> {
+    let content = std::fs::read_to_string("/etc/system-info.toml")
+        .context("fail to read /etc/system-info.toml")?;
+    Ok(toml::from_str::<SystemInfo>(&content)?)
+}
+
+fn bitcoind_version() -> Result<String> {
+    let output = std::process::Command::new("bitcoind")
+        .args(["--version"])
+        .output()
+        .context("could not run bitcoind command")?;
+    std::str::from_utf8(&output.stdout)?
+        .split('\n')
+        .next()
+        .and_then(|line| line.split("Bitcoin Core version ").nth(1))
+        .ok_or(anyhow!(
+            "failed to parse version from bitcoind command output"
+        ))
+        .map(|version| version.into())
+}
+
+fn cockroach_version() -> Result<String> {
+    let output = std::process::Command::new("cockroach")
```

Comment:
When I deploy this I don't see any output for the cockroach version. You need to call the cochroach-rpc wrapper.

Reply:
Got it, there are two wrappers `cochroach-rpc` and `cochroach-sql`.  It may be confused in getting `cockroach` version from `cockroach-rpc`, so the original binary is exposed and the version will directly get from there.

Reply:
Got it, there are two wrappers `cochroach-rpc` and `cochroach-sql`.  It may be confused in getting `cockroach` version from `cockroach-rpc`, so the original binary is exposed and the version will directly get from there.